.github/workflows/build-agent.yml

```diff
@@ -47,21 +48,29 @@ jobs:
       - name: Set github reference env
         run: echo ::set-env name=RELEASE_VERSION::${GITHUB_REF#refs/*/}
 
-      - name: build image
-        if: "!contains(github.ref, 'refs/tags/v')"
+      - name: build image 
+        if: ${{ github.event_name == 'pull_request' }} 
+        run: |
+          docker buildx build -f agent/Dockerfile.${{ matrix.dockerfile }} \
+            --tag shellhubio/agent:${{ github.sha }}-${{ matrix.dockerfile }} \
+            --platform linux/${{ matrix.arch }} .
```

Comment:
This and the below task seem the same. Aren't they?

Reply:
No, this step don't have "--push" option.

Reply:
Couldn't you have a docker push as a specific task?

Reply:
I can try, but my idea is keep it simple. The buildx execution would be responsible for everything. 
I will work to separate these steps